File: android/MyApplication5/app/src/main/java/com/example/myapplication/MainActivity.java
package com.example.myapplication;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

// AppCompatActivity 를 상속 받는 메인 Activity 클래스
public class MainActivity extends AppCompatActivity {

    // Activity 가 생성될 때 호출 되는 생명 주기 메서드
    @Override
    public void onCreate(Bundle savedInstanceState) { // Bundle savedInstanceState: 이전 상태 정보가 저장된 객체 (앱이 재시작될 때 사용)
        // 기본 적인 Activity 생성
        super.onCreate(savedInstanceState);
        // EdgeToEdge 기능 활성화 (앱 콘텐츠가 상태바와 네비게이션바 영역까지 확장되도록 설정 - API35 부터 적용)
        EdgeToEdge.enable(this);
        // 레이아웃 파일 연결 (R.layout.activity_main: activity_main.xml 파일과 연결)
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });
    }

    public void mOnClick(View v) {
        Intent intent = new Intent(this, SubActivity.class);
        startActivity(intent);
    }
}

File: android/ToDoList/app/src/main/java/com/logmind/todolist/MainActivity.kt
package com.logmind.todolist

import android.content.Intent
import android.os.Bundle
import android.widget.Toast
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import androidx.lifecycle.lifecycleScope
import androidx.recyclerview.widget.LinearLayoutManager
import com.logmind.todolist.databinding.ActivityMainBinding
import com.logmind.todolist.db.AppDatabase
import com.logmind.todolist.db.ToDoDao
import com.logmind.todolist.db.ToDoEntity
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch
import kotlinx.coroutines.withContext

class MainActivity : AppCompatActivity(), OnItemLongClickListener {

    private lateinit var binding: ActivityMainBinding

    private lateinit var db : AppDatabase
    private lateinit var todoDao: ToDoDao
    private lateinit var todoList : ArrayList<ToDoEntity>
    private lateinit var adapter : TodoRecyclerViewAdapter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }

        binding.btnAdd.setOnClickListener {
            val intent = Intent(this, AddTodoActivity::class.java)
            startActivity(intent)
        }

        db = AppDatabase.getInstance(this)!!
        todoDao = db.getTodoDao()

        getAllTodoList()
    }

    override fun onRestart() {
        super.onRestart()
        getAllTodoList()
    }

    private fun getAllTodoList () {
        lifecycleScope.launch {
            todoList = ArrayList(todoDao.getAll())
            withContext(Dispatchers.Main) {
                setRecyclerView()
            }
        }
    }

    private fun setRecyclerView() {
        adapter = TodoRecyclerViewAdapter(todoList, this)
        binding.recyclerView.adapter = adapter
        binding.recyclerView.layoutManager = LinearLayoutManager(this@MainActivity)
    }

    override fun onLongClick(position: Int) {
        val builder: AlertDialog.Builder = AlertDialog.Builder(this)
        builder.setTitle("할 일 삭제")
        builder.setMessage("정말 삭제하시겠습니까?")
        builder.setNegativeButton("취소", null)
        builder.setPositiveButton("네") { _, _ ->
            deleteTodo(position)
        }
        builder.show()
    }

    private fun deleteTodo(position: Int) {
        lifecycleScope.launch {
            todoDao.deleteTodo(todoList[position])
            todoList.removeAt(position)
            adapter.notifyDataSetChanged()
            Toast.makeText(this@MainActivity, "삭제되었습니다.", Toast.LENGTH_SHORT).show()
        }
    }
}